controller_interface/include/controller_interface/controller_state_names.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2021, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+/// \author: Denis Stogl
+
+#ifndef CONTROLLER_INTERFACE__CONTROLLER_STATE_NAMES_HPP_
+#define CONTROLLER_INTERFACE__CONTROLLER_STATE_NAMES_HPP_
+
+#include <string>
+
+namespace controller_interface
+{
+
+namespace state_names
+{
+/// Constants defining string labels corresponding to lifecycle states
+const auto UNCONFIGURED = "unconfigured";
+const auto INACTIVE = "inactive";
+const auto ACTIVE = "active";
+const auto FINALIZED = "finalized";
```

Comment:
nitpick, but `auto` here and include string is weird. Would auto here not default to char*?

You can put string or use https://en.cppreference.com/w/cpp/string/basic_string/operator%22%22s

Reply:
Yeah it should resolve to:

```
constexpr char UNCONFIGURED[] = "unconfigured";
```

Discussing this with @bmagyar, we would prefer:
```
const std::string UNCONFIGURED = "unconfigured";
```
but there are issues with `cpplint`.

@v-lopez Which version do you prefer?

Reply:
Constexpr char probably, our of curiosity what is the cpplint complain?

Reply:
That we should use that raw C definition (the first one above) instead of `std::string`

Reply:
Is it OK now?

Reply:
It is!